### PR TITLE
upcoming: [M3-7876] - Linode Create Refactor - Clone

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝

- Adds the ability to Clone in the new Linode Create flow ✨ 
- Improves validation when submitting form with no backup / linode selected

## Preview 📷

Screenshot 2024-04-30 at 10 33 24 AM

### Prerequisites
- Enable the `Linode Create v2` feature flag

### Verification steps
- Check the "Clone Linode" tab for parity with the existing create flow 🔍 

## As an Author I have considered 🤔

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [x] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/features/Linodes/LinodeCreatev2/resolvers.ts
Comment (bnussman-akamai) [inline packages/manager/src/features/Linodes/LinodeCreatev2/resolvers.ts:1]: There will probably be ways to consolidate and reduce duplication here, but I'm going to keep it explicit like this for now. As I become more familiar with react-hook-form, I'm sure I'll discover better ways to do things regarding our payload transformations and validators.

File: packages/manager/src/features/Linodes/LinodeCreatev2/Tabs/Backups/LinodeSelectTable.tsx
Comment (hkhalil-akamai) [inline packages/manager/src/features/Linodes/LinodeCreatev2/Tabs/Backups/LinodeSelectTable.tsx:1]: Maybe we should move this component to reflect that it's used for both Clone and Backups?

File: packages/manager/src/features/Linodes/LinodeCreatev2/Tabs/Backups/LinodeSelectTable.tsx [hunk 1]
```diff
@@ -126,46 +143,60 @@ export const LinodeSelectTable = () => {
                 >
                   Region
                 </TableSortCell>
+                {enablePowerOff && <TableCell></TableCell>}
```
Comment (hkhalil-akamai) [inline packages/manager/src/features/Linodes/LinodeCreatev2/Tabs/Backups/LinodeSelectTable.tsx:146]:
```suggestion
                {enablePowerOff && <TableCell />}
```

File: packages/manager/src/features/Linodes/LinodeCreatev2/Tabs/Backups/LinodeSelectTableRow.tsx [hunk 2]
```diff
@@ -47,13 +49,20 @@ export const LinodeSelectTableRow = (props: Props) => {
...
       </TableCell>
       <TableCell>{image?.label ?? linode.image}</TableCell>
       <TableCell>
         {type?.label ? formatStorageUnits(type.label) : linode.type}
       </TableCell>
       <TableCell>{region?.label ?? linode.region}</TableCell>
+      {onPowerOff && (
+        <TableCell actionCell>
+          {selected && linode.status !== 'offline' && (
+            <InlineMenuAction actionText="Power Off" onClick={onPowerOff} />
```
Comment (hkhalil-akamai) [inline packages/manager/src/features/Linodes/LinodeCreatev2/Tabs/Backups/LinodeSelectTableRow.tsx:62]: Minor visual nit: can we remove the small margins around this button to match what we have in prod?

| Prod | This PR|
|------|-------|
|Screenshot 2024-05-01 at 3 39 41 PM|Screenshot 2024-05-01 at 3 39 22 PM|

**Review by hkhalil-akamai — APPROVED** [review]
Nice work on this and great job sharing components where possible. Left some minor feedback but otherwise, great work!

✅ Code review
✅ Manual testing
✅ CI passing

**Review by abailly-akamai — COMMENTED** [review]
Looks great! test flow worked as expected and did not find any functionality regression.

File: packages/manager/src/features/Linodes/LinodeCreatev2/Tabs/Backups/LinodeSelectTable.tsx [hunk 3]
```diff
@@ -126,46 +143,60 @@ export const LinodeSelectTable = () => {
...
+          pageSize={pagination.pageSize}
+        />
+      </Box>
+      {enablePowerOff && (
+        <PowerActionsDialog
+          action="Power Off"
+          isOpen={Boolean(linodeToPowerOff)}
+          linodeId={linodeToPowerOff?.id}
+          onClose={() => setLinodeToPowerOff(undefined)}
+        />
```
[the remark this answers is not in the document]
Comment (bnussman-akamai) [reply packages/manager/src/features/Linodes/LinodeCreatev2/Tabs/Backups/LinodeSelectTable.tsx:199]: @hkhalil-akamai Any context on no "Power Off" in card view?

File: packages/manager/src/features/Linodes/LinodeCreatev2/Tabs/Backups/LinodeSelectTable.tsx
Comment (bnussman-akamai) [reply packages/manager/src/features/Linodes/LinodeCreatev2/Tabs/Backups/LinodeSelectTable.tsx:1]: I'd be down. What would be a good place? 

`packages/manager/src/features/Linodes/LinodeCreatev2/shared/LinodeSelectTable.tsx`
`packages/manager/src/features/Linodes/LinodeCreatev2/components/LinodeSelectTable.tsx`
`packages/manager/src/features/Linodes/LinodeCreatev2/Tabs/shared/LinodeSelectTable.tsx`
`packages/manager/src/features/Linodes/LinodeCreatev2/Tabs/components/LinodeSelectTable.tsx`

Any recommendations @abailly-akamai ?

File: packages/manager/src/features/Linodes/LinodeCreatev2/Tabs/Backups/LinodeSelectTable.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (bnussman-akamai) [reply packages/manager/src/features/Linodes/LinodeCreatev2/Tabs/Backups/LinodeSelectTable.tsx:199]: The "Power Off" button wrapping / column width issues should be fixed in 7f27bad  (cc @hkhalil-akamai )

**Review by cpathipa — APPROVED** [review]

File: packages/manager/src/features/Linodes/LinodeCreatev2/Tabs/Backups/LinodeSelectTable.tsx
Comment (bnussman-akamai) [reply packages/manager/src/features/Linodes/LinodeCreatev2/Tabs/Backups/LinodeSelectTable.tsx:1]: Done in 0f761a0 🎉

**Pull request merged**